### PR TITLE
feat(argo-workflows): Support images property to workflow-controller-configmap for emissary executor

### PR DESCRIPTION
**Pull request by yu-croco**
This PR resolves https://github.com/argoproj/argo-helm/issues/1707. 🙋 

---

Signed-off-by: yu-croco <yu.croco@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/templates/controller/workflow-controller-config-map.yaml [hunk 1]
```diff
@@ -170,3 +170,6 @@ data:
     {{- with .Values.controller.retentionPolicy }}
     retentionPolicy: {{- toYaml . | nindent 6 }}
     {{- end }}
+    {{- with .Values.emissaryExecutor.images }}
+    images: {{- toYaml . | nindent 6 }}
+    {{- end }}
```
Comment (yu-croco) [inline charts/argo-workflows/templates/controller/workflow-controller-config-map.yaml:175]: ref: https://argoproj.github.io/argo-workflows/workflow-controller-configmap.yaml
image

File: charts/argo-workflows/values.yaml [hunk 2]
```diff
@@ -630,3 +630,12 @@ customArtifactRepository: {}
...
 #       key: password
+
+emissaryExecutor:
+  # -- The command/args for each image on workflow, needed when the command is not specified and the emissary executor is used.
+  ## See more: https://argoproj.github.io/argo-workflows/workflow-executors/#emissary-emissary
+  images: []
+  #  argoproj/argosay:v2:
+  #    cmd: [/argosay]
+  #  docker/whalesay:latest:
+  #    cmd: [/bin/bash]
```
Comment (yu-croco) [inline charts/argo-workflows/values.yaml:641]: I wonder if there is better name.... 🤔 

*original config is below.
ref: https://argoproj.github.io/argo-workflows/workflow-controller-configmap.yaml
image

Comment (jmeridth) [reply charts/argo-workflows/values.yaml:641]: maybe just `emissary`.  links are there in comments to give more info.  wdyt?

Comment (yu-croco) [reply charts/argo-workflows/values.yaml:641]: Hi @jmeridth , thank you for your opinion! I will adopt your idea, below. 🙋 
```yaml
emissary:
  # -- The command/args for each image on workflow, needed when the command is not specified and the emissary executor is used.
  ## See more: https://argoproj.github.io/argo-workflows/workflow-executors/#emissary-emissary
  images: []
  #  argoproj/argosay:v2:
  #    cmd: [/argosay]
  #  docker/whalesay:latest:
  #    cmd: [/bin/bash]
```

Comment (yu-croco) [reply charts/argo-workflows/values.yaml]: Hi @jmeridth , is there anything in particular I can do to get approved? 🙋

**Review by jmeridth — APPROVED** [review]

**Pull request merged**